contracts/src/v0.8/dev/ChainSpecificUtil.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {ArbSys} from "./vendor/@arbitrum/nitro-contracts/src/precompiles/ArbSys.sol";
+
+//@dev A library that abstracts out opcodes that behave differently across chains.
+//@dev The methods below return values that are pertinent to the given chain.
+//@dev For instance, ChainSpecificUtil.getBlockNumber() returns L2 block number in L2 chains
+library ChainSpecificUtil {
+    address private constant ARBSYS_ADDR =
+        address(0x0000000000000000000000000000000000000064);
+    ArbSys private constant ARBSYS = ArbSys(ARBSYS_ADDR);
+    uint256 private constant ARB_MAINNET_CHAIN_ID = 42161;
+    uint256 private constant ARB_GOERLI_TESTNET_CHAIN_ID = 421613;
+
+    function getBlockhash(uint256 blockNumber) internal view returns (bytes32) {
+        uint256 chainid = getChainID();
+        if (
+            chainid == ARB_MAINNET_CHAIN_ID ||
+            chainid == ARB_GOERLI_TESTNET_CHAIN_ID
+        ) {
+            if ((getBlockNumber() - blockNumber) > 256 || blockNumber >= getBlockNumber()) {
+                return "";
+            }
+            return ARBSYS.arbBlockHash(blockNumber);
+        }
+        return blockhash(blockNumber);
+    }
+
+    function getBlockNumber() internal view returns (uint256) {
+        uint256 chainid = getChainID();
+        if (
+            chainid == ARB_MAINNET_CHAIN_ID ||
+            chainid == ARB_GOERLI_TESTNET_CHAIN_ID
+        ) {
+            return ARBSYS.arbBlockNumber();
+        }
+        return block.number;
+    }
+
```

Comment:
This ChainSpecificUtil contract doesn't seem to have the L1 fees stuff, maybe we can add it in here to stay consistent with the other file we have?

Reply:
I had missed that we have an existing ChainSpecificUtil in v0.8. Will delete this file and switch over.